cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/utils/FilterUtils.java

```diff
@@ -111,4 +115,29 @@ public static List<List<String>> cartesianProduct(List<List<String>> orderedValu
             }
         }
     }
+
+    /**
+     * Parses {@link TimeRangeFilter} from spark options.
+     */
+    public static TimeRangeFilter parseSSTableTimeRangeFilter(Map<String, String> options)
+    {
+        if (!options.containsKey(SSTABLE_START_TIMESTAMP_MICROS) && !options.containsKey(SSTABLE_END_TIMESTAMP_MICROS))
+        {
+            return null;
+        }
+
+        long startTimestamp = MapUtils.getLong(options, SSTABLE_START_TIMESTAMP_MICROS, Long.MIN_VALUE);
+        if (!options.containsKey(SSTABLE_END_TIMESTAMP_MICROS))
+        {
+            return TimeRangeFilter.startingAt(startTimestamp);
+        }
```

Comment:
I do not think it is necessary. Both cases (the end time option exists or not) are already covered in `TimeRangeFilter.create`